packages/react/src/errorboundary.tsx

```diff
@@ -0,0 +1,124 @@
+import * as Sentry from '@sentry/browser';
+import * as hoistNonReactStatic from 'hoist-non-react-statics';
+import * as React from 'react';
+
+export const UNKNOWN_COMPONENT = 'unknown';
+
+export type ErrorBoundaryProps = {
+  showDialog?: boolean;
+  dialogOptions?: Sentry.ReportDialogOptions;
+  fallback?: React.ReactNode;
+  renderKey?: string | number;
+  fallbackRender?(fallback: {
+    error: Error | null;
+    componentStack: string | null;
+    resetError(): void;
+  }): React.ReactNode;
+  onError?(error: Error, componentStack: string): void;
+  onMount?(): void;
+  onReset?(error: Error | null, componentStack: string | null): void;
+  onUnmount?(error: Error | null, componentStack: string | null): void;
+};
+
+type ErrorBoundaryState = {
+  componentStack: string | null;
+  error: Error | null;
+};
+
+const INITIAL_STATE = {
+  componentStack: null,
+  error: null,
+};
+
+class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
+  public state: ErrorBoundaryState = INITIAL_STATE;
+
+  public componentDidCatch(error: Error, { componentStack }: React.ErrorInfo): void {
+    Sentry.captureException(error, { contexts: { react: { componentStack } } });
+    const { onError, showDialog, dialogOptions } = this.props;
+    if (onError) {
+      onError(error, componentStack);
+    }
+    if (showDialog) {
+      Sentry.showReportDialog(dialogOptions);
+    }
+    this.setState({ error, componentStack });
```

Comment:
The docs recommend using https://reactjs.org/docs/react-component.html#static-getderivedstatefromerror to update state after an error is thrown, but we wouldn't be able to capture the component stack. Maybe add a comment about this?

Reply:
Yes good point, I add a comment about it.